conjur/api.py

```diff
@@ -135,16 +135,21 @@ def list_resources(self):
             'account': self._account
         }
         params.update(self._default_params)
-
         json_response = invoke_endpoint(HttpVerb.GET, ConjurEndpoint.RESOURCES,
                                         params,
+                                        query=list_constraints,
                                         api_token=self.api_token,
                                         ssl_verify=self._ssl_verify).content
-
         resources = json.loads(json_response.decode('utf-8'))
-        resource_list = map(lambda resource: resource['id'], resources)
 
-        return list(resource_list)
+        # returns the result as a list of resource ids instead of the raw JSON only
+        # when the user does not provide `inspect` as one of their filters
+        if 'inspect' not in list_constraints:
+            # For each element (resource) in the resources sequence, we extract the resource id
+            resource_list = map(lambda resource: resource['id'], resources)
```

Comment:
As discussed on the PR, this is what exactly what a --qurery arg would me e.g.
`--query 'Resources[*].ResourceID`

Reply:
oh neat as in giving the user the ability to filter the actual response